pandas-stubs/core/groupby/groupby.pyi

```diff
@@ -235,8 +239,10 @@ class GroupBy(BaseGroupBy[NDFrameT]):
     @final
     @overload
     def size(self: GroupBy[Series]) -> Series[int]: ...
-    @overload  # return type depends on `as_index` for dataframe groupby
-    def size(self: GroupBy[DataFrame]) -> DataFrame | Series[int]: ...
+    @overload
+    def size(self: DataFrameGroupByGen[ByT, Literal[True]]) -> Series[int]: ...  # type: ignore[misc]
```

Comment:
Surprisingly here `mypy` throws the error that the DataFrameGroupByGen is not a supertype of DataFrameGroupByGen, I have ignored the error but open to a better solution.

Reply:
I think the solution here is to push `def size` down to the `DataFrameGroupBy` and `SeriesGroupBy` classes, and remove it from the `GroupBy` class.